src/Components/Import/Import.tsx

```diff
@@ -1,32 +1,178 @@
-import React, {useState} from 'react';
+import React, { useState } from "react";
+import ReactPlayer from "react-player";
+import {
+  isImage,
+  isVideo,
+  ErrorModal,
+  showError,
+  trimmedName,
+  Media,
+  AddMedia
+} from "..";
+import styles from "./Import.module.css";
+import character from "../../Media/character.png";
+import { Button } from "react-bootstrap";
 
-interface Person {
-  firstName: string;
-}
+type Props = {
+  setShow: React.Dispatch<React.SetStateAction<boolean>>;
+  medias: Media[];
+  setMedias: React.Dispatch<React.SetStateAction<Media[]>>;
+  removeFile: (index: number) => void;
+  addMedia: () => void;
+  addFile: (newMedia: Media[]) => void;
+};
 
-interface Props {
-  text: string;
-  ok?: boolean;
-  i?: number;
-  fn?:(bob: string) => string;
-  person: Person;
-}
+const Import: React.FC<Props> = ({
+  setShow,
+  medias,
+  setMedias,
+  removeFile,
+  addMedia,
+  addFile,
+}) => {
+  const [onDragState, setOnDragState] = useState<boolean>(false);
+  const [onDropState, setOnDropState] = useState<boolean>(false);
 
-interface TextNode {
-  text: string
-}
+  const mediaElement: (addFiles: Media[], file: File) => void = (
+    addFiles,
+    file: File
+  ) => {
```

Comment:
This approach is definitely the right one for displaying different types of content 👍 but I think it comes a bit too early in the process. Currently what you're doing is:
 1. Uploading a file (or multiple files) in AddMedia.tsx
 2. Check the file type to determine if it's an image or video
 3. Create the React element
 4. Append the element to an array (here `addFiles`, please rename!)
 5. Take the entire array and append it to any existing media elements (also called 'addFile', but this one is fine because it's a function)
 6. Display the element in the list
 
WDYT about the following instead:
 1. Uploading a file (or multiple files)
 2. Add each file to an array of uploaded files (including previously uploaded ones) eg. step 5 above
 3. In the render, loop through the list of files and pass each to a 'MediaTile' (or similar) component, which:
 4. Checks the file type to determine if the file is an image or a video
 5. Creates and displays the React Element

IMO, the second is a cleaner and more readable approach.

Reply:
I agree with this Julia! I started the project by doing the latter, however I encounter some problems that makes me attempted the first one. The reasons being:
- Creating and storing allows the application to load all images and videos. 
- I can then get the duration of the images and videos from the beginning
- and apply duration slots from templates to the right image/video (for example, if the templates says to put all videos as 10 seconds and each video uploaded is only 5 seconds, then we can take account to that)
- progress bar can show how long the videos are with the loaded duration and other informations
- There won't be any buffering while video is playing as we know that we have loaded all of them before
- not really a pro, but it shrinks the amount of code and reduce repetition of creating the same elements multiple times
- I look through applications like jpg-to-pdf converters and it seem like they always load all medias first, before previewing them

Would storing JSX elements in an array a bad coding practice? I can definitely change it back to the second approach :D

Reply:
Those are all really good reasons for using the approach you have 👍 
I still think it would be cleaner to create all the elements in one pass, based off the final `medias` array when you click 'CreateVideo'. So basically:
 - User uploads files (stored in array)
 - User clicks 'Create video'
 - Modal opens, maybe with a loading spinner? (or the button shows a loading spinner, and then opens the modal). 
 - During the loading/processing, you create the elements, calculate durations, apply template etc.
 - Once everything is loaded, video plays

Reply:
yep sounds good, implementing it right away :) Just to clarify, when a user clicks "create video", in order to load every image/video file in a `<img>` and `<video>` or `<ReactPlayer/>` tag, we will then store these react elements into in a `Media[]` as created before. Would this be okay?

Reply:
Yep, but that should be within the Modal component 👍 
The reason for only building the elements there is that it encapsulates the code and keeps all the 'modal' things in one place. For instance, if you then needed to create a second modal at some other point in time, you'd only have to worry about passing it a set of files, rather than an array of elements. The idea is to ensure that each main section of your App could standalone with minimal dependencies